libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/FeedbackMetadata.kt

```diff
@@ -0,0 +1,197 @@
+package com.mapbox.navigation.core.telemetry.events
+
+import android.location.Location
+import com.google.gson.Gson
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.internal.utils.toTelemetryLocations
+import com.mapbox.navigation.core.telemetry.LocationsCollector
+
+/**
+ * It's the wrapper of [FeedbackMetadata] that collect locations after instance of
+ * [FeedbackMetadataWrapper] is created and stop collecting in 2 possible cases:
+ * - locations buffer is full;
+ * - [FeedbackMetadataWrapper.get] is called.
+ *
+ * Notes:
+ * - if you need to serialize Feedback metadata, use for that [FeedbackMetadata] only (see [FeedbackMetadataWrapper.get]);
+ * - to collect as much data as possible preferably call [FeedbackMetadataWrapper.get] just before posting or storing feedback;
+ * - [FeedbackMetadataWrapper.get] can be invoked whenever it needs.
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+class FeedbackMetadataWrapper internal constructor(
+    private val sessionIdentifier: String,
+    private val driverModeIdentifier: String,
+    @FeedbackEvent.DriverMode private val driverMode: String,
+    private val driverModeStartTime: String,
+    private val rerouteCount: Int,
+    private val lastLocation: Point?,
+    private val locationEngineNameExternal: String,
+    private val percentTimeInPortrait: Int?,
+    private val percentTimeInForeground: Int?,
+    private val eventVersion: Int,
+    private val phoneState: PhoneState,
+    private val navigationStepData: NavigationStepData,
+    private val appMetadata: AppMetadata?,
+    private val locationsCollector: LocationsCollector,
+) {
+    private var bufferFlushed = false
+
+    private val locationsBefore = mutableListOf<Location>()
+    private val locationsAfter = mutableListOf<Location>()
+
+    private val locationsBufferListener =
+        LocationsCollector.LocationsCollectorListener { preEventLocations, postEventLocations ->
+            locationsBefore.addAll(preEventLocations)
+            locationsAfter.addAll(postEventLocations)
+            bufferFlushed = true
+        }
+
+    init {
+        locationsCollector.collectLocations(locationsBufferListener)
+    }
+
+    /**
+     * Provide [FeedbackMetadata] with locations that were collected after creating instance of
+     * [FeedbackMetadataWrapper].
+     */
+    fun get(): FeedbackMetadata {
```

Comment:
Should this be implemented as a getter property? 🤔 That way clients won't need the `get()`.

Reply:
they're working the same, no sense to change, it doesn't bring value

Reply:
Well I don't agree, if you call it from Kotlin you _get_ the `get` notation for free 😅